File: app/src/main/java/com/prey/events/retrieves/EventRetrieveDataMobile.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2019 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.events.retrieves;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.prey.PreyConfig;
import com.prey.PreyLogger;
import com.prey.PreyPhone;
import com.prey.events.manager.EventManager;

import org.json.JSONException;
import org.json.JSONObject;

public class EventRetrieveDataMobile {

    public  void execute(Context context, EventManager manager){
        JSONObject mobileSon = new JSONObject();
        try {
            ConnectivityManager cm = (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);
            NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
            String mobile_internet=null;
            if(activeNetwork!=null && activeNetwork.getType() == ConnectivityManager.TYPE_MOBILE){
                try {
                    mobile_internet=PreyPhone.getNetworkClass(context);
                    PreyConfig.getPreyConfig(context).setPreviousSsid(mobile_internet);
                } catch (Exception e) {
                }
            }
            mobileSon.put("mobile_internet", mobile_internet);
            PreyLogger.d("mobile_internet:"+mobile_internet);
        } catch (Exception e) {
        }
        manager.receivesData(EventManager.MOBILE, mobileSon);
    }
}

File: app/src/main/java/com/prey/events/retrieves/kotlin/EventRetrieveDataMobile.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.events.retrieves.kotlin

import android.content.Context
import android.net.ConnectivityManager
import com.prey.events.manager.kotlin.EventManager
import com.prey.kotlin.PreyConfig
import com.prey.kotlin.PreyLogger
import com.prey.kotlin.PreyPhone
import org.json.JSONObject

class EventRetrieveDataMobile {
    fun execute(context: Context, manager: EventManager) {
        val mobileSon = JSONObject()
        try {
            val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetwork = cm.activeNetworkInfo
            var mobile_internet: String? = null
            if (activeNetwork != null && activeNetwork.type == ConnectivityManager.TYPE_MOBILE) {
                try {
                    mobile_internet = PreyPhone.getInstance(context)!!.getNetworkClass(context)
                    PreyConfig.getInstance(context).setPreviousSsid(mobile_internet)
                } catch (e: Exception) {
                    PreyLogger.e("Error:" + e.message, e)
                }
            }
            mobileSon.put("mobile_internet", mobile_internet)
            PreyLogger.d("mobile_internet:$mobile_internet")
        } catch (e: Exception) {
            PreyLogger.e("Error:" + e.message, e)
        }
        manager.receivesData(EventManager.MOBILE, mobileSon)
    }
}